src/order/monotone.lean

```diff
@@ -539,39 +540,71 @@ end linear_order
 section preorder
 variables [preorder α]
 
-lemma monotone_nat_of_le_succ {f : ℕ → α} (hf : ∀ n, f n ≤ f (n + 1)) :
-  monotone f | n m h :=
+lemma forall_nat_ge_lt_imp_rel_of_forall_rel_succ (r : β → β → Prop) [is_trans β r]
```

Comment:
This is a pretty obscure name. What about `nat.rel_of_le_of_lt_of_forall_rel_succ`? The other ones can then be `nat.rel_of_le_of_le_of_forall_rel_succ`, `nat.rel_of_lt_of_forall_rel_succ`, `nat.rel_of_le_of_forall_rel_succ`.

Reply:
I renamed almost as you suggested, but adjusted the order of words to the actual order of arguments in the lemmas.